tests/test-callbacks.js

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2017, Intel Corporation.
+
+var assert = require("Assert.js");
+var cb = require('test_callbacks');
+
+var times = 0;
+function callback(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "a prop"),
+            "args matched (" + times + ")");
+    times++;
+}
+
+var list_count = 0;
+function callback_l1(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "list"),
+            "list args matched (" + list_count + ")");
+    list_count++;
+}
+
+function callback_l2(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "list"),
+            "list args matched (" + list_count + ")");
+    list_count++;
+}
+
+function callback_l3(arg1, arg2, arg3)
+{
+    assert((arg1 == "one" &&
+            arg2 == "two" &&
+            arg3 == "three" &&
+            this.prop == "list"),
+            "list args matched (" + list_count + ")");
+    list_count++;
+}
+
+var id = cb.addCallback(callback, {prop: "a prop"});
+cb.signalCallback(id, "one", "two", "three");
+cb.signalCallback(id, "one", "two", "three");
+cb.signalCallback(id, "one", "two", "three");
+
+setTimeout(function() {
+    assert(times === 3, "signaled 3 times");
+    cb.removeCallback(id);
+    cb.signalCallback(id, "one", "two", "three");
+    setTimeout(function() {
+        assert(times === 3, "removed callback not called");
+        cb.signalCallback(-1);
+        cb.signalCallback(-10);
+        cb.signalCallback(12345);
```

Comment:
Changing these to anything other than a number wouldn't really be testing the callback module, but rather the callback test module itself. These numbers are passed directly to the C API's so it would essentially "do nothing" because we would have to throw away the argument.